public_html/wp-content/plugins/pattern-directory/includes/class-rest-favorite-controller.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace WordPressdotorg\Pattern_Directory\Favorites_API;
+
+use function WordPressdotorg\Pattern_Directory\Favorite\{add_favorite, get_favorites, remove_favorite};
+use WP_Error, WP_REST_Server, WP_REST_Response;
+
+add_action( 'rest_api_init', __NAMESPACE__ . '\init' );
+
+/**
+ * Set up the endpoints for favoriting patterns.
+ */
+function init() {
+	register_rest_route(
+		'wporg/v1',
+		'pattern-favorites',
+		array(
+			'methods' => WP_REST_Server::READABLE,
+			'callback' => __NAMESPACE__ . '\get_items',
+			'permission_callback' => __NAMESPACE__ . '\permissions_check',
+		)
+	);
+
+	$args = array(
+		'id' => array(
+			'validate_callback' => function( $param, $request, $key ) {
+				return is_numeric( $param );
+			},
+		),
+	);
+	register_rest_route(
+		'wporg/v1',
+		'pattern-favorites',
+		array(
+			'methods' => WP_REST_Server::EDITABLE,
+			'callback' => __NAMESPACE__ . '\update_item',
```

Comment:
Minor nitpick: now that we are actually adding a new row when favoriting something, it would be more accurate to have this be `WP_REST_Server::CREATABLE` and the method be `create_item`